docs/gallery/plot_dynamictable_howto.py

```diff
@@ -584,46 +583,190 @@
     description='an example table',
-    columns=[col6, col6_ind, col6_ind_ind],
+    columns=[col7, col7_ind, col7_ind_ind],
 )
 
 ###############################################################################
 # Access the first row using the same syntax as before, except now a list of
 # lists is returned. You can then index the resulting list of lists to access
 # the individual elements.
 
-table_double_ragged_col[0, 'col6']  # returns [['a', 'b', 'c'], ['d', 'e', 'f']]
-table_double_ragged_col['col6'][0]  # same as line above
-table_double_ragged_col['col6'][0][1]  # returns ['d', 'e', 'f']
+table_double_ragged_col[0, 'col7']  # returns [['a', 'b', 'c'], ['d', 'e', 'f']]
+table_double_ragged_col['col7'][0]  # same as line above
 
 ###############################################################################
-# Accessing the column named 'col6' using square bracket notation will return
+# Accessing the column named 'col7' using square bracket notation will return
 # the top-level :py:class:`~hdmf.common.table.VectorIndex` for the column.
-# Accessing the column named 'col6' using dot notation will return the
+# Accessing the column named 'col7' using dot notation will return the
 # :py:class:`~hdmf.common.table.VectorData` object
 
-table_double_ragged_col['col6']  # returns col6_ind_ind
-table_double_ragged_col.col6  # returns col6
+table_double_ragged_col['col7']  # returns col7_ind_ind
+table_double_ragged_col.col7  # returns col7
+
+###############################################################################
+# Accessing data from a ``DynamicTable`` that contain references to rows of other ``DynamicTable`` objects
+# --------------------------------------------------------------------------------------------------------
+# By default, when
+# :py:meth:`DynamicTable.__getitem__ <hdmf.common.table.DynamicTable.__getitem__>`
+# and :py:meth:`DynamicTable.get <hdmf.common.table.DynamicTable.get>` are supplied
+# with an int, list of ints, numpy array, or a slice representing rows to return,
+# a pandas :py:class:`~pandas.DataFrame` is returned. If the
+# :py:class:`~hdmf.common.table.DynamicTable` contains a
+# :py:class:`~hdmf.common.table.DynamicTableRegion` column that references rows
+# of other ``DynamicTable`` objects, then by default, the
+# :py:meth:`DynamicTable.__getitem__ <hdmf.common.table.DynamicTable.__getitem__>`
+# and :py:meth:`DynamicTable.get <hdmf.common.table.DynamicTable.get>` methods will
+# return row indices of the referenced table, and not the contents of the referenced
+# table. To return the contents of the referenced table as a nested
+# :py:class:`~pandas.DataFrame` containing only the referenced rows, use
+# :py:meth:`DynamicTable.get <hdmf.common.table.DynamicTable.get>` with ``index=False``.
+#
+# You can also get rows from a :py:class:`~hdmf.common.table.DynamicTable` as a list of
+# lists where the i-th nested list contains the values for the i-th row. This method is
+# generally not recommended.
```

Comment:
I would change this to  ".. note::" instead since you are not recommending this behavior. It would also be useful to indicate briefly why this is not recommended.

Reply:
Done. Thanks.